### DISCUSSION TITLE
Using gum spin

### DISCUSSION THREAD

**Discussion by lbdyck** [charmbracelet/gum#477] [Q&A] [answered]
I've read about `gum spin` and tried it but my question is how can I start a process and have the `spin` active until that process completes?

Thanks

**Comment by maaslalani** [comment]
Hey @lbdyck,

`gum spin` takes as arguments the process to run and spins until it's complete. For example, you can run `sleep` (or any bash command).

```bash
gum spin -- sleep 5
```

More useful is probably the ability to run bash scripts

```bash
#!/bin/bash
# do whatever you want
sleep 5
```

```bash
chmod +x ./script.sh
gum spin -- ./script.sh
```

**Comment by lbdyck** [comment]
Nice - thank you very much - I need to review the doc more closely

**Comment by lbdyck** [comment]
I'm trying this with no joy:

` gum spin -s line --title="list"  man -k ssh`
and
` gum spin -s line --title="list"  man -k ssh | gum pager`

neither work telling me that the -k is invalid for `gum` ???

**Reply by maaslalani** [reply]
Hey @lbdyck,

I think you need to separate the command with a `--` so the `-k` option isn't interpreted by gum:

```
gum spin -s line --title="list" --  man -k ssh | gum pager
```

**Reply by lbdyck** [reply]
Tried that as well with no joy :(

![image](https://github.com/charmbracelet/gum/assets/42328411/229b9127-8c8c-4384-ac79-5c416f9b22fa)

**Reply by lbdyck** [reply]
This is what I would expect to see in pager

![image](https://github.com/charmbracelet/gum/assets/42328411/0e7fa9b5-10ab-4750-a0a9-a8bc66f638d2)

**Reply by lbdyck** [reply]
And it does appear in pager if I remove the `gum spin`

**Reply by maaslalani** [reply]
Ah apologies, I missed the `| gum pager`. I don't think it's possible to combine `--` and `|` since it will not be interpreted correctly by `gum spin`.

**Reply by lbdyck** [reply]
ok - so no way to have a spinner for a long running command to feed `gum pager` or `gum choose` ?

**Reply by maaslalani** [reply, marked as answer]
You can do something like this:

```
gum spin man ssh --show-output | gum pager
```

I tested this and it works for me. Note the `--show-output` to pipe into `gum pager`

**Reply by lbdyck** [reply]
Awesome - THANK YOUI

**Comment by lbdyck** [comment]
But `gum spin man -k ssh --show-output | gum choose` fails and that is what I was really hoping for since the search can talk awhile.
